lib/fluent/plugin/input.rb

```diff
@@ -29,6 +29,27 @@ class Input < Base
 
       helpers_internal :event_emitter
 
+      def initialize
+        super
+        @emit_records = 0
+        @emit_size = 0
+        @counter_mutex = Mutex.new
+      end
+
+      def statistics
+        stats = {
+          'emit_records' => @emit_records,
+          'emit_size' => @emit_size,
+        }
+
+        { 'input' => stats }
+      end
+
+      def metric_callback(es)
+        @counter_mutex.synchronize { @emit_records += 1 }
+        @counter_mutex.synchronize { @emit_size += es.to_msgpack_stream.bytesize }
```

Comment:
Maybe I'm missing, does it need to be separated `synchronize` ?

Reply:
I guess it should be better to minimize synchronize regions.

Reply:
By separating `synchronize`, `emit_records` and `emit_size` may have inconsistent value.

Reply:
Oh, I see....